exp/webhooks/ipaddress_test.go

```diff
@@ -0,0 +1,216 @@
+				Kind:     "TestPool",
+				Name:     "pool",
+				APIGroup: pointer.String("ipam.cluster.x-k8s.io"),
+			},
+		},
+	}
+
+	getAddress := func(v6 bool, fn func(addr *expv1.IPAddress)) expv1.IPAddress {
+		addr := expv1.IPAddress{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+			},
+			Spec: expv1.IPAddressSpec{
+				ClaimRef: corev1.LocalObjectReference{Name: claim.Name},
+				PoolRef:  claim.Spec.PoolRef,
+				Address:  "10.0.0.1",
+				Prefix:   24,
+				Gateway:  "10.0.0.254",
+			},
+		}
+		if v6 {
+			addr.Spec.Address = "42::1"
+			addr.Spec.Prefix = 64
+			addr.Spec.Gateway = "42::ffff"
+		}
+		fn(&addr)
+		return addr
+	}
+
+	tests := []struct {
+		name      string
+		ip        expv1.IPAddress
+		extraObjs []client.Object
+		expectErr bool
+	}{
+		{
+			name:      "a valid IPv4 Address should be accepted",
+			ip:        getAddress(false, func(addr *expv1.IPAddress) {}),
+			extraObjs: []client.Object{claim},
+			expectErr: false,
+		},
+		{
+			name:      "a valid IPv6 Address should be accepted",
+			ip:        getAddress(true, func(addr *expv1.IPAddress) {}),
+			extraObjs: []client.Object{claim},
+			expectErr: false,
+		},
+		{
+			name: "a prefix that is too large should be rejected",
+			ip: getAddress(false, func(addr *expv1.IPAddress) {
+				addr.Spec.Prefix = 64
+			}),
+			extraObjs: []client.Object{claim},
+			expectErr: true,
+		},
+		{
+			name: "an invalid address should be rejected",
+			ip: getAddress(false, func(addr *expv1.IPAddress) {
+				addr.Spec.Address = "42"
+			}),
+			extraObjs: []client.Object{claim},
+			expectErr: true,
+		},
+		{
+			name: "an invalid gateway should be rejected",
+			ip: getAddress(false, func(addr *expv1.IPAddress) {
+				addr.Spec.Gateway = "42"
+			}),
+			extraObjs: []client.Object{claim},
+			expectErr: true,
+		},
+		{
+			name: "a reference to a non-existent claim should be rejected",
+			ip: getAddress(false, func(addr *expv1.IPAddress) {
+				addr.Spec.ClaimRef.Name = "nothing"
+			}),
+			extraObjs: []client.Object{claim},
+			expectErr: true,
+		},
+		{
+			name: "a pool reference that does not match the claim should be rejected",
+			ip: getAddress(false, func(addr *expv1.IPAddress) {
+				addr.Spec.PoolRef.Name = "nothing"
+			}),
+			extraObjs: []client.Object{claim},
+			expectErr: true,
+		},
+		{
+			name: "a pool reference that does not contain a group should be rejected",
+			ip: getAddress(false, func(addr *expv1.IPAddress) {
+				addr.Spec.PoolRef.APIGroup = nil
+			}),
+			extraObjs: []client.Object{claim},
+			expectErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			scheme := runtime.NewScheme()
+			g.Expect(expv1.AddToScheme(scheme)).To(Succeed())
+			wh := IPAddress{
+				Client: fake.NewClientBuilder().WithScheme(scheme).WithObjects(tt.extraObjs...).Build(),
+			}
+			if tt.expectErr {
+				g.Expect(wh.validate(context.Background(), &tt.ip)).NotTo(Succeed())
```

Comment:
Let's call ValidateCreate, just gives us slightly more test coverage and works out-of-the-box 

(same in other cases)